ui/src/shared/components/RangeThresholdMarkers.tsx

```diff
@@ -0,0 +1,66 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+import {Scale} from '@influxdata/giraffe'
+
+// Components
+import ThresholdMarker from 'src/shared/components/ThresholdMarker'
+import ThresholdMarkerArea from 'src/shared/components/ThresholdMarkerArea'
+
+// Utils
+import {isInDomain, clamp} from 'src/shared/utils/vis'
+import {DragEvent} from 'src/shared/utils/useDragBehavior'
+
+// Types
+import {RangeThresholdConfig} from 'src/types'
+
+interface Props {
+  yScale: Scale<number, number>
+  yDomain: number[]
+  threshold: RangeThresholdConfig
+  onChangeMaxPos: (e: DragEvent) => void
+  onChangeMinPos: (e: DragEvent) => void
+}
+
+const RangeThresholdMarkers: FunctionComponent<Props> = ({
+  yScale,
+  yDomain,
+  threshold: {level, within, minValue, maxValue},
+  onChangeMinPos,
+  onChangeMaxPos,
+}) => {
+  const minY = clamp(yScale(minValue), yDomain)
+  const maxY = clamp(yScale(maxValue), yDomain)
+
+  return (
+    <>
+      {within ? (
+        <ThresholdMarkerArea level={level} top={maxY} height={minY - maxY} />
+      ) : (
+        <>
+          {maxValue <= yDomain[1] && (
+            <ThresholdMarkerArea
+              level={level}
+              top={yScale(yDomain[1])}
+              height={maxY - yScale(yDomain[1])}
+            />
+          )}
+          {minValue >= yDomain[0] && (
+            <ThresholdMarkerArea
+              level={level}
+              top={minY}
+              height={yScale(yDomain[0]) - minY}
+            />
+          )}
+        </>
+      )}
+      {isInDomain(minValue, yDomain) && (
```

Comment:
I stared at this for a while before parsing that this was outside the ternary, can the markers be above the marker areas please?

Reply:
jeez, almost like you're a human instead of jsx parser! I'll update it